### PR TITLE
Update the CI versions

### PR DESCRIPTION
**Pull request by raimis**
- [x] Include PyTorch 1.12
- [x] Update from the source installation

### CODE REVIEW

**Comment by raimis** [comment]
@peastman could you review?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,30 +15,46 @@ defaults:
...
+          - name: Linux (CUDA 11.2, Python 3.9, PyTorch 1.11)
+            enable_cuda: true
+            cuda: "11.2.2"
+            gcc: "10.3.*"
+            nvcc: "11.2"
+            python: "3.9.*"
+            pytorch: "1.11.*"
+
+          # Latest supported versions (with CUDA)
+          - name: Linux (CUDA 11.2, Python 3.10, PyTorch 1.12)
```
Comment (peastman) [inline .github/workflows/ci.yml:44]: All of these are older versions.  How about CUDA 11.8, Python 3.11, and PyTorch 1.13?

Comment (raimis) [reply .github/workflows/ci.yml]: Python 3.11 and PyTorch 1.13 are not available because there are no corresponding TorchANI packages.

Comment (raimis) [reply .github/workflows/ci.yml]: Changed to CUDA 11.7. CUDA 11.8 isn't available because of the `cuda-toolkit` action not yet supporting it.

**Review by peastman — APPROVED** [review]
Looks good.

**Pull request merged**